[ROCm] /horovod/common/operations.cc:1808: Error: `status` not declared during build

**Environment:**
1. Framework: PyTorch
2. Framework version: v1.12.1_src
3. Horovod version: v0.27.0
4. MPI version: N/A
5. ROCm version: 5.1.1
6. NCCL version: N/A
7. Python version: 3.9.16
8. Spark / PySpark version: N/A
9. Ray version: N/A
10. OS and version:
11. GCC version: 10.3.0
12. CMake version: 3.22.2

**Checklist:**
1. Did you search issues to find if somebody asked this question before?
2. If your question is about hang, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/running.rst)?
3. If your question is about docker, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/docker.rst)?
4. Did you check if you question is answered in the [troubleshooting guide](https://github.com/horovod/horovod/blob/master/docs/troubleshooting.rst)?

**Bug report:**
Please describe erroneous behavior you're observing and steps to reproduce it.

Building Horovod v0.27.0 with PyTorch-ROCm produces the following error:

```
[ 79%] Building CXX object horovod/torch/CMakeFiles/pytorch.dir/__/common/optim/gaussian_process.cc.o
cd /home/xu.3304/horovod/build/temp.linux-x86_64-cpython-39/RelWithDebInfo/horovod/torch && /opt/gcc/10.3.0/bin/g++ -DEIGEN_MPL2_ONLY=1 -DHAVE_GLOO=1 -DHAVE_GPU=1 -DHAVE_NCCL=1 -DHAVE_ROCM=1 -DHOROVOD_GPU_ALLGATHER=78 -DHOROVOD_GPU_ALLREDUCE=78 -DHOROVOD_GPU_ALLTOALL=78 -DHOROVOD_GPU_BROADCAST=78 -DHOROVOD_GPU_REDUCESCATTER=78 -DPYTORCH_VERSION=1012000000 -DTORCH_API_INCLUDE_EXTENSION_H=1 -D__HIP_PLATFORM_AMD__=1 -D__HIP_PLATFORM_HCC__=1 -Dpytorch_EXPORTS -I/home/xu.3304/horovod/third_party/HTTPRequest/include -I/home/xu.3304/horovod/third_party/boost/assert/include -I/home/xu.3304/horovod/third_party/boost/config/include -I/home/xu.3304/horovod/third_party/boost/core/include -I/home/xu.3304/horovod/third_party/boost/detail/include -I/home/xu.3304/horovod/third_party/boost/iterator/include -I/home/xu.3304/horovod/third_party/boost/lockfree/include -I/home/xu.3304/horovod/third_party/boost/mpl/include -I/home/xu.3304/horovod/third_party/boost/parameter/include -I/home/xu.3304/horovod/third_party/boost/predef/include -I/home/xu.3304/horovod/third_party/boost/preprocessor/include -I/home/xu.3304/horovod/third_party/boost/static_assert/include -I/home/xu.3304/horovod/third_party/boost/type_traits/include -I/home/xu.3304/horovod/third_party/boost/utility/include -I/home/xu.3304/horovod/third_party/lbfgs/include -I/home/xu.3304/horovod/third_party/gloo -I/home/xu.3304/horovod/third_party/eigen -I/home/xu.3304/horovod/third_party/flatbuffers/include -isystem /opt/rocm-5.1.1/include -isystem /opt/rocm-5.1.1/rccl/include -isystem /home/xu.3304/miniconda3/envs/torch1.12.1-hvd0.27.0/lib/python3.9/site-packages/torch/include -isystem /home/xu.3304/miniconda3/envs/torch1.12.1-hvd0.27.0/lib/python3.9/site-packages/torch/include/torch/csrc/api/include -isystem /home/xu.3304/miniconda3/envs/torch1.12.1-hvd0.27.0/lib/python3.9/site-packages/torch/include/TH -isystem /home/xu.3304/miniconda3/envs/torch1.12.1-hvd0.27.0/lib/python3.9/site-packages/torch/include/THC -isystem /home/xu.3304/miniconda3/envs/torch1.12.1-hvd0.27.0/lib/python3.9/site-packages/torch/include/THH -isystem /opt/rocm-5.1.1/miopen/include -isystem /opt/rocm-5.1.1/hip/include -isystem /home/xu.3304/miniconda3/envs/torch1.12.1-hvd0.27.0/include/python3.9 -DCUDA_HAS_FP16=1 -D__HIP_NO_HALF_OPERATORS__=1 -D__HIP_NO_HALF_CONVERSIONS__=1 -D_GLIBCXX_USE_CXX11_ABI=1 -D__HIP_PLATFORM_HCC__=1  -pthread -fPIC -Wall -ftree-vectorize -mf16c -mavx -mfma -O3 -g -DNDEBUG -fPIC -std=c++14 -MD -MT horovod/torch/CMakeFiles/pytorch.dir/__/common/optim/gaussian_process.cc.o -MF CMakeFiles/pytorch.dir/__/common/optim/gaussian_process.cc.o.d -o CMakeFiles/pytorch.dir/__/common/optim/gaussian_process.cc.o -c /home/xu.3304/horovod/horovod/common/optim/gaussian_process.cc
/home/xu.3304/horovod/horovod/common/operations.cc: In function ‘horovod::common::Status horovod::common::EnqueueTensorReducescatters(std::vector<std::shared_ptr<horovod::common::OpContext> >&, std::vector<std::shared_ptr<horovod::common::Tensor> >&, std::vector<horovod::common::ReadyEventList>&, std::vector<std::__cxx11::basic_string<char> >&, int, std::vector<std::function<void(const horovod::common::Status&)> >&, horovod::common::ReduceOp, int32_t, double, double)’:
/home/xu.3304/horovod/horovod/common/operations.cc:1808:12: error: ‘status’ was not declared in this scope; did you mean ‘Status’?
 1808 |     return status.Aborted("AVERAGE not allowed.");
      |            ^~~~~~
      |            Status
```
The error is pretty straightforward, it happens at `horovod/common/operations.cc`, the function `EnqueueTensorReducescatters` doesn't declare a `status`

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -16,6 +16,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ### Fixed
 
+- Fixed build with ROCm. ([#3839](https://github.com/horovod/horovod/pull/3839))
+
 
 ## [v0.27.0] - 2023-02-01
 

--- a/horovod/common/operations.cc
+++ b/horovod/common/operations.cc
@@ -1805,7 +1805,7 @@ EnqueueTensorReducescatters(std::vector<std::shared_ptr<OpContext>>& contexts,
 #else
     LOG(ERROR, horovod_global.global_controller->GetRank())
         << "Enqueuing AVERAGE reducescatter is not allowed with ROCm.";
-    return status.Aborted("AVERAGE not allowed.");
+    return Status::Aborted("AVERAGE not allowed.");
 #endif
     break;
   case ReduceOp::SUM:

